### PR TITLE
2671: Clone draft functionality with react query

### PR DESCRIPTION
**Pull request by SaadAli-rolustech**
**What does this PR?**
-clone button added on draft list item
- user can clone a draft

 **Issue number**
https://github.com/ecency/ecency-mobile/issues/2671

 **Screenshots/Video**

https://user-images.githubusercontent.com/39906620/235132628-1780b581-343d-4dc0-9f65-430fe242aef2.mov

### CODE REVIEW

**Review by noumantahir — CHANGES_REQUESTED** [review]
Great job on this PR @SaadAli-rolustech

As discussed in meeting, couple changes required.....

- [x] use filled copy instead of bordered or change both delete and copy to bordered icons
- [x] instead of creating clone mutation separate, make on addDraftMutation that we can used later in editorContainer as well.

**Comment by SaadAli-rolustech** [comment]
Thank you for the feedback. The PR has been updated, Please review it.

File: src/providers/ecency/ecency.ts [hunk 1]
```diff
@@ -116,20 +116,22 @@ export const deleteDraft = async (draftId: string) => {
...
+  const { title, body, tags, meta, action } = draft;
   try {
     const data = { title, body, tags, meta };
     const res = await ecencyApi.post('/private-api/drafts-add', data);
-    const { drafts } = res.data;
-    if (drafts) {
-      return drafts.pop(); //return recently saved last draft in the list
-    } else {
-      throw new Error('No drafts returned in response');
+    if (action === 'add') {
```
Comment (noumantahir) [inline src/providers/ecency/ecency.ts:126]: using exclusive property `action` may be unnecessary and confusing for later use... why not just return the full response data array, `editorContainer` can just post process to extract last draft item....

**Review by noumantahir — CHANGES_REQUESTED** [review]
looking good, just a tinee tiny change for code cleaning....

File: src/providers/ecency/ecency.ts [hunk 2]
```diff
@@ -116,21 +116,14 @@ export const deleteDraft = async (draftId: string) => {
...
   try {
     const data = { title, body, tags, meta };
     const res = await ecencyApi.post('/private-api/drafts-add', data);
-    const { drafts } = res.data;
-    if (drafts) {
-      return drafts.pop(); //return recently saved last draft in the list
-    } else {
-      throw new Error('No drafts returned in response');
-    }
+    return res.data || [];
```
Comment (noumantahir) [inline src/providers/ecency/ecency.ts:126]: please test addDraft response again, it should return array of drafts... res.data do not contain the expected array....
https://media.discordapp.net/attachments/1094029149892509767/1103977166485598218/Screenshot_2023-05-05_at_2.30.31_PM.png

**Review by noumantahir — CHANGES_REQUESTED** [review]
please test both cloning and saving new draft from editor screen

**Review by noumantahir — APPROVED** [review]
wrapped up PR, fixed addDraft response processing

**Pull request merged**